scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/activities/ActivityExpression.java

```diff
@@ -614,4 +618,84 @@ public String prettyPrint(final String prefix) {
   @Override
   public void extractResources(final Set<String> names) { }
 
+
+  /**
+   * Evaluates whether a SerializedValue can be qualified as the subset of another SerializedValue or not
+    * @param arguments the proposed superset
+   * @param pattern the proposed subset
+   * @return true if pattern is a subset of arguments
+   */
+  public static boolean subsetOrEqual(SerializedValue arguments, SerializedValue pattern){
+    final var visitor = new SerializedValue.Visitor<Boolean>(){
+      @Override
+      public Boolean onNull() {
+        return true;
+      }
+
+      @Override
+      public Boolean onNumeric(final BigDecimal value) {
+        final var argumentsAsNumeric = arguments.asNumeric();
+        if(argumentsAsNumeric.isEmpty()){
+          return false;
+        }
+        return argumentsAsNumeric.get().equals(value);
+      }
+
+      @Override
+      public Boolean onBoolean(final boolean value) {
+        final var argumentsAsBoolean = arguments.asBoolean();
+        if(argumentsAsBoolean.isEmpty()){
+          return false;
+        }
+        return argumentsAsBoolean.get().equals(value);
+      }
+
+      @Override
+      public Boolean onString(final String value) {
+        final var argumentsAsString = arguments.asString();
+        if(argumentsAsString.isEmpty()){
+          return false;
+        }
+        return argumentsAsString.get().equals(value);
+      }
+
+      @Override
+      public Boolean onMap(final Map<String, SerializedValue> value) {
+        final var argumentsAsMap = arguments.asMap();
+        if(argumentsAsMap.isEmpty()){
+          return false;
+        }
+        for(final var elementInPattern: value.entrySet()){
+          final var elementInArguments = argumentsAsMap.get().get(elementInPattern.getKey());
+          if(elementInArguments != null){
+            if(!subsetOrEqual(elementInArguments, elementInPattern.getValue())){
+              return false;
+            }
+          } else {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      @Override
+      public Boolean onList(final List<SerializedValue> value) {
+        final var optionalArgumentsAsList = arguments.asList();
```

Comment:
This is minor, feel free to resolve, but `argumentsAsListOptional` is clearer than `optionalArgumentsAsList`.

Reply:
done